### PR TITLE
feat: account invite friends page

### PR DESCRIPTION
**Pull request by sshanzel**
Blocked: profile menu PR to add a link to this page

TODO:
- [x] include the link in the profile menu
- [x] UserList UX improvement

## Changes
- New component: SocialShareList to reuse the icons to share as they have the same behaviors.
- Fix the title "Other Settings" to lowercase the "Settings" part.
- Invite friends page with referred users list.

Previews:

Screenshot 2023-11-10 at 4 03 00 PM
Screenshot 2023-11-10 at 4 02 51 PM
Screenshot 2023-11-10 at 4 02 40 PM
Screenshot 2023-11-10 at 3 55 52 PM
Screenshot 2023-11-10 at 3 55 39 PM
Screenshot 2023-11-10 at 3 55 26 PM

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1928 #done

### CODE REVIEW

File: packages/webapp/components/layouts/AccountLayout/common.tsx [hunk 1]
```diff
@@ -117,12 +119,22 @@ export const accountPage: Record<AccountPage, AccountPageProps> = {
...
+      />
+    ),
+  },
 };
 
 export const AccountPageContent = classed(
   'div',
   pageBorders,
-  'flex flex-col w-full laptop:max-w-[calc(100vw-19.75rem)] laptop:max-w-[calc(100vw-36rem)] laptopL:max-w-[40rem] tablet:border-l mr-auto',
+  'flex flex-col w-full laptop:max-w-[calc(100vw-19.75rem)] laptopL:max-w-[40.5rem] tablet:border-l mr-auto',
```
Comment (sshanzel) [inline packages/webapp/components/layouts/AccountLayout/common.tsx:137]: There was an 8px discrepancy after checking in Sketch. Also removed the unused class.

File: packages/webapp/pages/account/invite.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        title="Referred"
+        description="Here you will see all your active members who joined to daily.dev. Who knows? maybe someday it will be worth something 😉"
+      >
+        <UserList
+          users={usersResult.data?.pages.reduce((acc, p) => {
+            p?.referredUsers.edges.forEach(({ node }) => {
+              acc.push(node);
+            });
+
+            return acc;
```
Comment (rebelchris) [inline packages/webapp/pages/account/invite.tsx:106]: Should we not do this reduce outside the render?

File: packages/shared/src/components/widgets/SocialShareList.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            <CopyIcon
+              className={classNames('text-theme-label-invert')}
+              secondary={isCopying}
+            />
+          }
+          className="btn-primary"
+          label={isCopying ? 'Copied!' : 'Copy link'}
+        />
+      )}
+      <SocialShareIcon
```
Comment (rebelchris) [inline packages/shared/src/components/widgets/SocialShareList.tsx:57]: The sizes match of the icons between the two implementations?
Didn't check, but the invite page one seem bigger then I recall (48x48px)

**Review by rebelchris — APPROVED** [review]
Nice work, only non blocking comments from my side.

File: packages/webapp/pages/account/invite.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/webapp/pages/account/invite.tsx:109]: Yep. Was actually thinking of creating a hook here, seems to be reused in numerous places.

File: packages/shared/src/components/widgets/SocialShareList.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/widgets/SocialShareList.tsx:57]: Yes, they are 48x48 in both cases.

**Comment by sshanzel** [comment]
Actual preview:

image

**Pull request merged**